Author: Kai-Chung Yan <seamlikok@gmail.com>
Last-Update: 2016-07-20
Description: Adopt the latest ANTLR APIs
 compilerCommon misuses some of the ANTLR APIs. This patch fixes:
   * Usage of the constructor of org.antlr.v4.runtime.atn.LexerATNSimulator
   * Override abstract getATN() of org.antlr.v4.runtime.Recognizer
--- a/compilerCommon/src/main/xml-gen/android/databinding/parser/XMLParser.java
+++ b/compilerCommon/src/main/xml-gen/android/databinding/parser/XMLParser.java
@@ -8,6 +8,8 @@
 import java.util.List;
 
 public class XMLParser extends Parser {
+	@Override
+	public ATN getATN() { return _ATN; }
 	public static final int
 		OPEN=7, CDATA=2, SLASH=13, CharRef=5, SEA_WS=6, SPECIAL_CLOSE=11, CLOSE=10, 
 		DTD=3, Name=16, EQUALS=14, PI=18, S=17, SLASH_CLOSE=12, TEXT=9, COMMENT=1, 
@@ -39,7 +41,7 @@
 
 	public XMLParser(TokenStream input) {
 		super(input);
-		_interp = new ParserATNSimulator(this,_ATN);
+		_interp = new ParserATNSimulator(this, _ATN, new DFA[10], new PredictionContextCache());
 	}
 	public static class DocumentContext extends ParserRuleContext {
 		public ElementContext element() {
--- a/compilerCommon/src/main/xml-gen/android/databinding/parser/XMLLexer.java
+++ b/compilerCommon/src/main/xml-gen/android/databinding/parser/XMLLexer.java
@@ -8,6 +8,8 @@
 import org.antlr.v4.runtime.atn.LexerATNSimulator;
 
 public class XMLLexer extends Lexer {
+	@Override
+	public ATN getATN() { return _ATN; }
 	public static final int
 		COMMENT=1, CDATA=2, DTD=3, EntityRef=4, CharRef=5, SEA_WS=6, OPEN=7, XMLDeclOpen=8, 
 		TEXT=9, CLOSE=10, SPECIAL_CLOSE=11, SLASH_CLOSE=12, SLASH=13, EQUALS=14, 
@@ -33,7 +35,7 @@
 
 	public XMLLexer(CharStream input) {
 		super(input);
-		_interp = new LexerATNSimulator(this,_ATN);
+		_interp = new LexerATNSimulator(this, _ATN, new org.antlr.v4.runtime.dfa.DFA[10], new org.antlr.v4.runtime.atn.PredictionContextCache());
 	}
 
 	@Override
--- a/compilerCommon/src/main/grammar-gen/android/databinding/parser/BindingExpressionParser.java
+++ b/compilerCommon/src/main/grammar-gen/android/databinding/parser/BindingExpressionParser.java
@@ -10,6 +10,8 @@
 import java.util.ArrayList;
 
 public class BindingExpressionParser extends Parser {
+	@Override
+	public ATN getATN() { return _ATN; }
 	public static final int
 		T__0=1, T__1=2, T__2=3, T__3=4, T__4=5, T__5=6, T__6=7, T__7=8, T__8=9, 
 		T__9=10, T__10=11, T__11=12, T__12=13, T__13=14, T__14=15, T__15=16, T__16=17, 
@@ -97,7 +99,7 @@
 
 	public BindingExpressionParser(TokenStream input) {
 		super(input);
-		_interp = new ParserATNSimulator(this,_ATN);
+		_interp = new ParserATNSimulator(this, _ATN, new DFA[10], new PredictionContextCache());
 	}
 	public static class BindingSyntaxContext extends ParserRuleContext {
 		public BindingSyntaxContext(ParserRuleContext parent, int invokingState) {
--- a/compilerCommon/src/main/grammar-gen/android/databinding/parser/BindingExpressionLexer.java
+++ b/compilerCommon/src/main/grammar-gen/android/databinding/parser/BindingExpressionLexer.java
@@ -10,6 +10,8 @@
 import org.antlr.v4.runtime.misc.*;
 
 public class BindingExpressionLexer extends Lexer {
+	@Override
+	public ATN getATN() { return _ATN; }
 	public static final int
 		T__0=1, T__1=2, T__2=3, T__3=4, T__4=5, T__5=6, T__6=7, T__7=8, T__8=9, 
 		T__9=10, T__10=11, T__11=12, T__12=13, T__13=14, T__14=15, T__15=16, T__16=17, 
@@ -99,7 +101,7 @@
 
 	public BindingExpressionLexer(CharStream input) {
 		super(input);
-		_interp = new LexerATNSimulator(this,_ATN);
+		_interp = new LexerATNSimulator(this, _ATN, new DFA[10], new PredictionContextCache());
 	}
 
 	@Override
